tests/integration/gcp_test.go

```diff
@@ -0,0 +1,82 @@
+/*
+Copyright 2022 The Flux authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package integration
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	tfjson "github.com/hashicorp/terraform-json"
+
+	tftestenv "github.com/fluxcd/image-reflector-controller/tests/tftestenv"
+)
+
+// createKubeconfigGKE constructs kubeconfig from the terraform state output at
+// the given kubeconfig path.
+func createKubeconfigGKE(ctx context.Context, state map[string]*tfjson.StateOutput, kcPath string) error {
+	kubeconfigYaml, ok := state["gcp_kubeconfig"].Value.(string)
+	if !ok || kubeconfigYaml == "" {
+		return fmt.Errorf("failed to obtain kubeconfig from tf output")
+	}
+
+	f, err := os.Create(kcPath)
+	if err != nil {
+		return err
+	}
+	_, err = fmt.Fprint(f, kubeconfigYaml)
+	return err
```

Comment:
Same here, we need to close the file handle `f.Close()` and check for errors as well:

```golang
	if err := f.Close(); err != nil {
            return err
	}
```